plugins/poll/assets/javascripts/discourse/components/poll-buttons-dropdown.gjs

```diff
@@ -0,0 +1,145 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import icon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+
+const buttonOptionsMap = {
+  exportResults: {
+    className: "btn-default export-results",
+    label: "poll.export-results.label",
+    title: "poll.export-results.title",
+    icon: "download",
+    action: "exportResults",
+  },
+  showBreakdown: {
+    className: "btn-default show-breakdown",
+    label: "poll.breakdown.breakdown",
+    icon: "chart-pie",
+    action: "showBreakdown",
+  },
+  openPoll: {
+    className: "btn-default toggle-status",
+    label: "poll.open.label",
+    title: "poll.open.title",
+    icon: "unlock-alt",
+    action: "toggleStatus",
+  },
+  closePoll: {
+    className: "btn-default toggle-status",
+    label: "poll.close.label",
+    title: "poll.close.title",
+    icon: "lock",
+    action: "toggleStatus",
+  },
+};
+
+export default class PollButtonsDropdownComponent extends Component {
+  @service currentUser;
+  @service siteSettings;
+  @tracked getDropdownButtonState;
+
+  constructor() {
+    super(...arguments);
+    this.getDropdownButtonState = false;
+  }
+  @action
+  toggleDropdownButtonState() {
+    this.getDropdownButtonState = !this.getDropdownButtonState;
+  }
+
+  @action
+  dropDownClick(dropDownAction) {
+    this.toggleDropdownButtonState();
+    this.args.dropDownClick(dropDownAction);
+  }
+
+  get dropDownButtonState() {
+    return this.getDropdownButtonState ? "opened" : "closed";
+  }
+
+  get getDropdownContent() {
+    const contents = [];
+    const isAdmin = this.currentUser && this.currentUser.admin;
+
+    const dataExplorerEnabled = this.siteSettings.data_explorer_enabled;
+    const exportQueryID = this.siteSettings.poll_export_data_explorer_query_id;
+
+    const closed = this.args.closed;
+    const voters = this.args.voters;
+    const isStaff = this.args.isStaff;
+    const isMe = this.args.isMe;
+    const topicArchived = this.args.topicArchived;
+    const groupableUserFields = this.args.groupableUserFields;
+    const isAutomaticallyClosed = this.args.isAutomaticallyClosed;
```

Comment:
Could make use of destructuration?

```suggestion
    const { 
      closed,
      voters,
      isStaff,
      isMe,
      topicArchived,
      groupableUserFields,
      isAutomaticallyClosed
    } = this.args;
```

Reply:
nice suggestion, committed: https://github.com/discourse/discourse/pull/27155/commits/c1fe372cce2729b989fdd59ec0446626e12f3d8b